recipe/meta.yaml

```diff
@@ -1,26 +1,34 @@
-{% set version = "2.6.0" %}
-{% set sha256 = "7c425211a099f6fa6f74037e6e17be58fb5923b0bd11aea745e48ef83c488b49" %}
+{% set version = "3.0.0" %}
 
 package:
   name: singularity
   version: {{ version }}
 
 source:
-  fn: singularity-{{ version }}.tar.gz
-  url: https://github.com/sylabs/singularity/releases/download/{{ version }}/singularity-{{ version }}.tar.gz
-  sha256: {{ sha256 }}
+  - folder: src/github.com/sylabs/singularity
+    url: https://github.com/sylabs/singularity/releases/download/v{{ version }}/singularity-v{{ version }}.tar.gz
+    sha256: 6d54daf74748018082e71774f0998a20d8242b8bda5d2c7d42573242556230a0
+    patches:
+      - 0001-use-c-ldflags-during-mconfig
```

Comment:
@jakirkham, I am not sure I am the best person to reach out to them about their new make scripts. 

Essentially, I would like to avoid a similar situation that occurred with the java/maven and conda a handful of months ago.

Reply:
Not a problem. Have raised issue ( https://github.com/sylabs/singularity/issues/2454 ).

Mainly we are just interested in conveying to upstream that there is a problem and how we addressed it. In the early days of c-f, we pushed a bit harder on getting patches submitted in PRs. However this is difficult to do in practice for a number of reasons (i.e. source may not be in version control, the process of submitting patches can sometimes be complex technically and/or legally, upstream may not be maintained (legacy software packaging for dependencies), etc.).

Agree this is significantly harder with the Java ecosystem (especially for JDK) not only because some issues above may come into play, but there is the additional problem of many OpenJDKs, which complicates the package management process further.

Reply:
Instead of creating a patch for their mconfig, I created a [wrapper to our compilers](https://github.com/sodre/singularity-feedstock/blob/d40654ff1e0eaacff88a5b2f5931b95aa369d95a/recipe/build.sh#L10-L21) on the fly inside build.sh and told mconfig to use that instead. 

My reasoning is that this should be easier to maintain versus a diff patch.